x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/prebuilt_rules/management/trial_license_complete_tier/upgrade_review_prebuilt_rules.ts

```diff
@@ -0,0 +1,679 @@
+
+            // Call the upgrade review prebuilt rules endpoint and check that one rule is eligible for update
+            // and single line string field update has conflict
+            const reviewResponse = await reviewPrebuiltRulesToUpgrade(supertest);
+            expect(reviewResponse.rules[0].diff.fields).toMatchObject({
+              name: {
+                base_version: 'A',
+                current_version: 'B',
+                diff_outcome: ThreeWayDiffOutcome.CustomizedValueCanUpdate,
+                has_conflict: true,
+                has_update: true,
+                merge_outcome: ThreeWayMergeOutcome.Conflict,
+                merged_version: 'B',
+                target_version: 'C',
+              },
+              version: {
+                base_version: 1,
+                current_version: 1,
+                diff_outcome: ThreeWayDiffOutcome.StockValueCanUpdate,
+                has_conflict: false,
+                has_update: true,
+                merge_outcome: ThreeWayMergeOutcome.Target,
+                merged_version: 2,
+                target_version: 2,
+              },
+            });
+            expect(reviewResponse.rules[0].diff.has_conflict).toBe(true);
+            expect(reviewResponse.stats.num_rules_to_upgrade_total).toBe(1);
+          });
+        });
+
+        describe('when rule base version does not exist', () => {
+          describe('when rule field has an update and a custom value that are the same', () => {
+            it('should not show in the upgrade/_review API response', async () => {
+              // Install all prebuilt detection rules
+              const ruleAssetSavedObjects = getRuleAssetSavedObjects();
+              await createPrebuiltRuleAssetSavedObjects(es, ruleAssetSavedObjects);
+              await installPrebuiltRules(es, supertest);
+
+              // Clear previous rule assets
+              await deleteAllPrebuiltRuleAssets(es, log);
+
+              // Customize a single line string field on the installed rule
+              await patchRule(supertest, log, {
+                rule_id: 'rule-1',
+                name: 'B',
+              });
+
+              // Increment the version of the installed rule, update a single line string field, and create the new rule assets
+              const updatedRuleAssetSavedObjects = [
+                createRuleAssetSavedObject({
+                  rule_id: 'rule-1',
+                  version: 2,
+                  name: 'B',
+                }),
+              ];
+              await createPrebuiltRuleAssetSavedObjects(es, updatedRuleAssetSavedObjects);
+
+              // Call the upgrade review prebuilt rules endpoint and check that one rule is eligible for update
+              // but does NOT contain single line string field
+              const reviewResponse = await reviewPrebuiltRulesToUpgrade(supertest);
+              expect(reviewResponse.rules[0].diff.fields).toMatchObject({
```

Comment:
Do we also want to add an explicit check that `name` field is not among `fields`? As I understand, `.toMatchObject` would still pass even if `name` field was there.

Maybe something like
```ts
expect(reviewResponse.rules[0].diff.fields).not.toHaveProperty('name');
```

Also, please check if similar tests need to be updated as well.

Reply:
Good catch, replacing with `toEqual` 👍